src/application/general.rs

```diff
@@ -68,6 +69,12 @@ impl<T> ApplicationExt<'_, T> for Application<T> {
         let window_attributes = WindowAttributes::from(&attributes);
         window_builder.window = window_attributes;
         let window = window_builder.build(&self.event_loop)?;
+        if attributes.x.is_some() && attributes.y.is_some() {
```

Comment:
When you use `is_some()` and then `unwrap()` inside the `if` condition, you're doing the same thing twice, which is checking whether the `Option` is `Some` or `None`. This would be more idiomatic if you wrote it like this:

```rust
match (attributes.x, attributes.y) {
    (Some(x), Some(y)) => {
		// set window properties here
	}
    _ => {}
}
```

Reply:
I didn't write that, it was part of #22. I used that as a base so it showed up here.

Reply:
Thanks for the great advice, I barely know rust so my most of the time my code won't be idiomatic.

Reply:
most of the time my code isn't idiomatic either LOL some code reviews just destroy me